SNOW-787480: fix json load encoding error (#1528)
snowflakedb/snowflake-connector-python: DESCRIPTION.md

# Release Notes

- v3.0.4(Unreleased)

>   - Fixed a bug that JSON formatted returning data is decoded in wrong encoding.

Reviewer: ```suggestion
  - Fixed a bug where query results data in JSON format is decoded with the wrong encoding.
```

Reviewer: Shall we link https://github.com/chardet/chardet/issues/148?
